server/aws-lsp-codewhisperer/src/language-server/chat/chatController.ts

```diff
@@ -200,24 +200,42 @@ export class ChatController implements ChatHandlers {
             return
         }
 
-        const cursorPosition = params.cursorPosition
+        let cursorPosition = params.cursorPosition
 
         const indentRange = {
             start: { line: cursorPosition.line, character: 0 },
             end: cursorPosition,
         }
         const documentContent = await this.#features.workspace.getTextDocument(params.textDocument.uri)
         let indent = documentContent?.getText(indentRange)
+
+        let hasVirtualSpace = false
         if (indent && indent.trim().length !== 0) {
             indent = ' '.repeat(indent.length - indent.trimStart().length)
+        } else if ((isNullish(indent) || !indent) && cursorPosition.character > 0) {
+            // When text content at indent range is undefined or empty but the cursor is not at position 0
+            // It means there are virtual spaces that is being rendered by the IDE
+            // In this case, the indentation is determined by the cursorPosition
+            this.#log('Indent is nullish and the cursor position is greater than zero while inserting code')
+            indent = ' '.repeat(cursorPosition.character)
+            hasVirtualSpace = true
+            cursorPosition.character = 0
         }
 
         let textWithIndent = ''
         params.code.split('\n').forEach((line, index) => {
             if (index === 0) {
```

Comment:
nit: this code will trim the first newline in case the code starts with `/n/n/n`

Reply:
In general, this code becomes difficult to read with nested conditions. I would think about how to flatten it out (like, move `if(!textWithIndent) {textWithIndent+='\n' + indent}` to top) or split to functions for different pre-conditions (like, with virtual spaces and without).